### ISSUE TITLE
Touch and resize issues on mobile android chome

### ISSUE THREAD

**Issue by CrazyFlasher** [haxeui/haxeui-heaps#24]
Using this sample
https://github.com/haxeui/component-examples/tree/master/buttons

http://188.166.108.195/testui/ - build

1. Touches doesn't work
2. Content is shown only in landscape mode
3. After rotating device - weird scaling

Scaling content issues can be easily reproduced on desktop. Just resize the window.

Took latest git of heaps, haxeui-code and haxeui-heaps
Build script:
```
-cp src
-main Main

-lib haxeui-core
-lib haxeui-heaps
-lib heaps

-js build/html5/Main.js
```

**Comment by ianharrigan** [comment]
Is heaps supported in mobile browers? My understanding was that heaps wasnt a mobile framework... but maybe thats inaccurate?

**Comment by CrazyFlasher** [comment]
https://heaps.io/samples/index.html
Yes, it is supported

**Comment by ianharrigan** [comment]
Right, so i guess ill have to test it a little... the scaling one is fairly easy, its because the canvas in your test app is 100%x100%... so i suppose it makes sense that its scaled all weird... not sure a good way around that... 

I mean, if you take one of those heaps samples and make it full page 100%x100% then i guess they will also behave strangely wrt to scale? 

Although interestingly, when i look at them in fullscreen: 

https://heaps.io/samples/adjustColor/

or

https://heaps.io/samples/pbr/

And resize the browser things look alright... i can see heaps stuttering though so it looks like its doing _something_ ill have to take a look and see - but it wont be till the weekend most likely since it doesnt look like a 5 second job

Cheers,
Ian

**Comment by CrazyFlasher** [comment]
> I mean, if you take one of those heaps samples and make it full page 100%x100% then i guess they will also behave strangely wrt to scale?

Scaling is fine for samples. For ex. check this sample:
http://188.166.108.195/heaps/build/base2D/

> And resize the browser things look alright... i can see heaps stuttering though so it looks like its doing something ill have to take a look and see - but it wont be till the weekend most likely since it doesnt look like a 5 second job

thanks!

**Comment by CrazyFlasher** [comment]
I've build heaps samples from github and got the same scaling issue:
![image](https://user-images.githubusercontent.com/1607138/125068868-a4e7e100-e0be-11eb-8c85-5bad1a720d70.png)

The html are exactly the same,
Howerver on their site resizing page works fine. I've report the issue in heaps repo
https://heaps.io/samples/base2D/

**Comment by CrazyFlasher** [comment]
I've tried different scaleModes in Base3D init method - but behavior is always the same.
Html source:
```html
<html>
<head><meta charset="utf-8"/><title>input</title></head>
<body style="margin:0;padding:0;background-color:black">
	
	<canvas id="webgl" style="width:100%;height:100%"></canvas>
	<script type="text/javascript" src="input.js"></script>
	
</body>
</html>
```

**Comment by CrazyFlasher** [comment]
![image](https://user-images.githubusercontent.com/1607138/125073007-09f20580-e0c4-11eb-990e-06737bde5acf.png)
for some reason, onResize is not called.
I've debugged sample here https://heaps.io/samples/base2D/ and onResize executes during stage resize

**Comment by CrazyFlasher** [comment]
I guess I found a commit "Remove useless listener and timer" db3ddc34385e132344c7e3f242ae114f061c0554 :)
![image](https://user-images.githubusercontent.com/1607138/125075117-debce580-e0c6-11eb-9c8f-0313f0c94174.png)
Looks like timer did the trick

**Comment by CrazyFlasher** [comment]
I found what causes the issue and it is heaps related:
https://github.com/HeapsIO/heaps/issues/986

**Comment by MSGhero** [comment]
My touch input is working on JS (iPhone Safari browser), but the x,y of the touch is always 0,0. This is because `Window.getInstance().mouseX/Y` are not updated on touch. `event.relX/Y` are what is updated. I would have to mess with scaling and stuff before just committing this, as that may not be the right numbers to report to HaxeUI.

**Comment by ianharrigan** [comment]
Interesting, so could relX / relY be use "across the board" (ie, even on desktop)

**Comment by ianharrigan** [comment]
Seems to work fine, yeah... i mean, its relative, but its relative to the root (which is usually s2d), so i guess thats fine, and the same as window absolute i would guess.
